modules/rostests/apitests/win32nt/dlihook.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * PROJECT:     ReactOS delayimport Library
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Delay-import hook for win32u.dll
+ * COPYRIGHT:   Copyright 2025 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#include <windows.h>
+#include <delayimp.h>
+#include <stdio.h>
+#include <versionhelpers.h>
+
+static
+FARPROC
+WINAPI
+DliNotifyHook(
+    unsigned code,
+    PDelayLoadInfo pdli)
+{
+    if (code != dliNotePreLoadLibrary)
+    {
+        return NULL;
+    }
+
+    if (strcmp(pdli->szDll, "win32u.dll") != 0)
+    {
+        return NULL;
+    }
+
+    if (IsReactOS())
+    {
+        return NULL;
+    }
+
+    switch (_winver)
+    {
+        case _WIN32_WINNT_WINXP:
+            pdli->szDll = "win32u_xpsp2.dll";
+            break;
+
+        case _WIN32_WINNT_WS03:
+            pdli->szDll = "win32u_2k3sp2.dll";
+            break;
+
+        case _WIN32_WINNT_VISTA:
+            pdli->szDll = "win32u_vista.dll";
+            break;
+
+        default:
+            break;
```

Comment:
```suggestion
        default:
            printf("Unsupported _winver = 0x%x\n", _winver);
            break;
```

Reply:
No, they are supported.

Reply:
`"Using ReactOS win32u.dll on Windows 7+ (_winver = 0x%x)\n"`, then.

Reply:
It's literally 2 lines below.